File: src/desktop/View/Qt/Defense/defense_view_factory.py
from Defense.anti_drone_mine import AntiDroneMine
from Defense.anti_drone_turret import AntiDroneTurret
from Defense.mine import Mine
from Defense.fragile_mine import FragileMine
from Defense.reverse_mine import ReverseMine
from Defense.worm import Worm

from View.Qt.Defense.anti_drone_mine_view import AntiDroneMineView
from View.Qt.Defense.anti_drone_turret_view import AntiDroneTurretView
from View.Qt.Defense.mine_view import MineView
from View.Qt.Defense.fragile_mine_view import FragileMineView
from View.Qt.Defense.reverse_mine_view import ReverseMineView
from View.Qt.Defense.worm_view import WormView

DefenseToViewDictionary = {AntiDroneMine:AntiDroneMineView,
                           AntiDroneTurret:AntiDroneTurretView,
                           Mine:MineView,
                           FragileMine:FragileMineView,
                           ReverseMine:ReverseMineView,
                           Worm:WormView}
                        
def GetViewForDefense(defense, parentView):
    """  """
    if defense.__class__ in DefenseToViewDictionary:
        return DefenseToViewDictionary[defense.__class__](defense, parentView)
    else:
        return MineView(defense, parentView)
    

File: src/android/src/com/kaottessur/pymine/view/defense/helper/DefenseViewFactory.java
package com.kaottessur.pymine.view.defense.helper;

import com.kaottessur.pymine.defense.AntiDroneTurret;
import com.kaottessur.pymine.defense.DefenseInterface;
import com.kaottessur.pymine.defense.FragileMine;
import com.kaottessur.pymine.defense.Mine;
import com.kaottessur.pymine.defense.ReverseMine;
import com.kaottessur.pymine.defense.Worm;
import com.kaottessur.pymine.level.Drone;
import com.kaottessur.pymine.level.Minefield;
import com.kaottessur.pymine.view.defense.AntiDroneTurretSprite;
import com.kaottessur.pymine.view.defense.DefenseSprite;
import com.kaottessur.pymine.view.defense.FragileMineSprite;
import com.kaottessur.pymine.view.defense.MineSprite;
import com.kaottessur.pymine.view.defense.ReverseMineSprite;
import com.kaottessur.pymine.view.defense.WormSprite;

public class DefenseViewFactory {
	
	public static DefenseSprite getDefenseView(DefenseInterface defense, Drone drone, Minefield minefield) {
		if (defense.getClass() == Mine.class) {
			return new MineSprite((Mine) defense);
		} else if (defense.getClass() == ReverseMine.class) {
			return new ReverseMineSprite((Mine) defense);
		} else if (defense.getClass() == FragileMine.class) {
			return new FragileMineSprite((Mine) defense);
		} else if (defense.getClass() == Worm.class) {
			return new WormSprite((Worm) defense, drone, minefield);
		} else if (defense.getClass() == AntiDroneTurret.class) {
			return new AntiDroneTurretSprite((AntiDroneTurret) defense);
		}
		return null;
	}
	
}
